prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -193,7 +193,17 @@ class Hl7Serializer(
             }
 
             if (it.isNotBlank()) {
-                nextMessage.append("$it\r")
+                if (it.length + nextMessage.length <= nextMessage.capacity()) {
+                    if (it.length > 4000) {
+                        logger.warn("Message segment ${it.substring(0,3)} longer than 4000 characters, truncating")
+                        nextMessage.append("${it.substring(0,4000)}\r")
```

Comment:
same as above, magic number? If you change this 4000 in the future we should only have to change it in one spot.